### DISCUSSION TITLE
React-Hook-Form Input File submit value

### DISCUSSION THREAD

**Discussion by ibrahimbdr** [react-hook-form/react-hook-form#12449] [Q&A]
Hey Devs,
I have a question in react-hook-form where I have a file-input inside react hook form and I am using custom input like:

import {Controller} from 'react-hook-form';

        <Controller
          name={props.name}
          control={control}
          render={({ field: { onChange } }) => (
            <FileInput {...props} onChange={onChange} />
          )}
        />

and inside the FileInput

I have my rendered input defined as:

        <input
          {...register(name)} // Registering the input with react-hook-form
          type="file"
          id={name}
          className={inputClasses}
          onChange={handleUpload} // Handling file upload on change
        />

when submitting the form, I am getting value for the input as "[object FileList]" of type string

Do any one knows how resolve this issue?

**Comment by evgeniyworkbel** [comment]
Hi! "[object FileList]" means that field value is coercing to string. Can you show how you get value on submit?
